### DISCUSSION TITLE
Mapped domain stores are not belong to this domain.

### DISCUSSION THREAD

**Discussion by abliarsar** [zerobias/effector#294] [Ideas]
**What is the current behavior:**
Stores created with map are always created in global scope

**Please provide the steps to reproduce and if possible a minimal demo of the problem**
```javascript
const domain = createDomain()
domain.createStore(1).map(a => 2)
console.log(domain.history.stores.size) // => 1
```
[repl](https://share.effector.dev/DJEHFqzt)

**What is the expected behavior:**
Mapped store has the same domain as his parent.
Or a parameter to choose which domain derived store should belong to.

**Which versions of Effector packages, and which browser and OS are affected by this issue? Did this work in previous versions of Effector?:**
All

**Comment by zerobias** [comment]
Yes, derived stores aren't presented in domains, as it is pretty dangerous to use nearly every api method with them: by making `store.map` we assuming that source store will be single source of truth for derived one thereby using api methods like `forward` or `on` with it inside domain hooks leads to unexpected behavior.

Therefore, I wonder how it is supposed to use derived stores in domains?

**Comment by abliarsar** [comment]
Hmm, I thought it would be useful at least for logging. Right now, the only way to do something with mapped stores is manually replacing .map method in onCreateStore hook
[effector-logger issue](https://github.com/sergeysova/effector-logger/issues/2)

**Comment by abliarsar** [comment]
Another possible use case is shallow fork, which we discussed earlier in telegram. Mapped stores should be copied, but right now we can't tell whether this store is derived or just regular global one

**Comment by zerobias** [comment]
Mapped stores will be forked anyway, because semantically they are belong to their parent store. For example, for that reason using `.on` with them considered bad practice. Truly independent stores can be connected together via `.on`:

```js
const source = createStore(0)

const independentTarget = createStore(0)
  .on(source, (_, source) => source * 10)

const dependentTarget = source.map(source => source * 10)
```
[Try it](https://share.effector.dev/2gqB2RKd)

In this example in case of shallow fork `dependentTarget` will be forked and `independentTarget` will stay the same
